components/service/glean/src/main/java/mozilla/components/service/glean/ping/BaselinePing.kt

```diff
@@ -69,4 +72,45 @@ internal class BaselinePing(applicationContext: Context) {
             it.id
         }
     }
+
+    /**
+     * Gets a gecko-compatible locale string (e.g. "es-ES" instead of Java [Locale]
+     * "es_ES") for the default locale.
+     *
+     * This method approximates the API21 method [Locale.toLanguageTag].
+     *
+     * @return a locale string that supports custom injected languages.
+     */
+    internal fun getLanguageTag(): String {
+        // Thanks to toLanguageTag() being introduced in API21, we could have
+        // simple returned `locale.toLanguageTag();` from this function. However
+        // what kind of languages the Android build support is up to the manufacturer
+        // and our apps usually support translations for more rare languages, through
+        // our custom locale injector. For this reason, we can't use `toLanguageTag`
+        // and must try to replicate its logic ourselves.
+        val locale = Locale.getDefault()
+        val language = getLanguage(locale)
+        val country = locale.country // Can be an empty string.
+        return if (country.isEmpty()) language else "$language-$country"
+    }
+
+    /**
+     * Sometimes we want just the language for a locale, not the entire language
+     * tag. But Java's .getLanguage method is wrong.
+     *
+     * @param locale a [Locale] object to be stringified.
+     * @return a language string, such as "he" for the Hebrew locales.
+     */
+    internal fun getLanguage(locale: Locale): String {
+        // Can, but should never be, an empty string.
+        val language = locale.language
+
+        // Modernize certain language codes.
+        return when (language) {
+            "iw" -> "he"
+            "in" -> "id"
+            "ji" -> "yi"
+            else -> language
+        }
```

Comment:
I assume this mapping is brought over from desktop...?  Do we want to reference it so we don't loose track of why this is needed?

Reply:
This whole code was ported over from the legacy Telemetry library that lives in a-c. I should have mentioned that, sorry. Here's the [original code](https://github.com/mozilla-mobile/android-components/blob/345f45448311d8c64f89e9c2a1fb435d2d46d853/components/service/telemetry/src/main/java/org/mozilla/telemetry/measurement/LocaleMeasurement.java#L48-L65). I've updated the docstring to mention that the mapping is available in `toLanguageTag` docs.